wallet/src/main/java/com/android/identity/issuance/WalletDocumentMetadata.kt

```diff
@@ -0,0 +1,167 @@
+package com.android.identity.issuance
+
+import com.android.identity.cbor.annotation.CborSerializable
+import com.android.identity.document.DocumentMetadata
+import com.android.identity.issuance.remote.WalletServerProvider
+import kotlinx.coroutines.sync.Mutex
+import kotlinx.coroutines.sync.withLock
+import kotlinx.io.bytestring.ByteString
+import kotlinx.io.bytestring.isEmpty
+import kotlin.concurrent.Volatile
+
+/**
+ * [DocumentMetadata] implementation for the Wallet app.
+ */
+class WalletDocumentMetadata private constructor(
+    serializedData: ByteString?,
+    private val saveFn: suspend (data: ByteString) -> Unit
+) : DocumentMetadata {
+    private val lock = Mutex()
```

Comment:
Some data access is protected by this lock, and some isn't. Most of the getters aren't, which means they could get data that's in the middle of being changed (eg. may not have been save()d yet). Are we sure that's safe?

Reply:
k, sounds good. Thanks.